docs/tutorials/csharp/bert-nlp-csharp-console-app.md

```diff
@@ -0,0 +1,376 @@
+  // Get the sentence tokens.
+  var tokens = tokenizer.Tokenize(sentence);
+  // Console.WriteLine(String.Join(", ", tokens));
+
+  // Encode the sentence and pass in the count of the tokens in the sentence.
+  var encoded = tokenizer.Encode(tokens.Count(), sentence);
+
+  // Break out encoding to InputIds, AttentionMask and TypeIds from list of (input_id, attention_mask, type_id).
+  var bertInput = new BertInput()
+  {
+      InputIds = encoded.Select(t => t.InputIds).ToArray(),
+      AttentionMask = encoded.Select(t => t.AttentionMask).ToArray(),
+      TypeIds = encoded.Select(t => t.TokenTypeIds).ToArray(),
+  };
+ 
+```
+### Create the Tensors
+- Create the `ConvertToTensor` function. Set the shape of the Tensor `new[] { 1, inputDimension }` and the values to be added to the `NamedOnnxValue` input list.
+
+```csharp
+        public static Tensor<long> ConvertToTensor(long[] inputArray, int inputDimension)
+        {
+            // Create a tensor with the shape the model is expecting. Here we are sending in 1 batch with the inputDimension as the amount of tokens.
+            Tensor<long> input = new DenseTensor<long>(new[] { 1, inputDimension });
+
+            // Loop through the inputArray (InputIds, AttentionMask and TypeIds)
+            for (var i = 0; i < inputArray.Length; i++)
+            {
+                // Add each to the input Tenor result.
+                // Set index and array value of each input Tensor.
+                input[0,i] = inputArray[i];
+            }
+            return input;
+        }
+```
+
+### Create the `input` of `List<NamedOnnxValue>` that is needed for inference
+
+- Get the model, call the `ConvertToTensor` function to create the tensor and create the list of `NamedOnnxValue` input variables for inferencing.
+
+```csharp
+  // Get path to model to create inference session.
+  var modelPath = @"C:\code\bert-nlp-csharp\BertNlpTest\BertNlpTest\bert-large-uncased-finetuned-qa.onnx";
+
+  // Create input tensor.
+
+  var input_ids = ConvertToTensor(bertInput.InputIds, bertInput.InputIds.Length);
+  var attention_mask = ConvertToTensor(bertInput.AttentionMask, bertInput.InputIds.Length);
+  var token_type_ids = ConvertToTensor(bertInput.TypeIds, bertInput.InputIds.Length);
+
+
+  // Create input data for session.
+  var input = new List<NamedOnnxValue> { NamedOnnxValue.CreateFromTensor("input_ids", input_ids), 
+                                         NamedOnnxValue.CreateFromTensor("input_mask", attention_mask), 
+                                         NamedOnnxValue.CreateFromTensor("segment_ids", token_type_ids) };
+
+
+```
+### Run Inference
+- Create the `InferenceSession`, run the inference and print out the result.
+
+```csharp
+  // Create an InferenceSession from the Model Path.
+  var session = new InferenceSession(modelPath);
+
+  // Run session and send the input data in to get inference output. 
+  var output = session.Run(input);
+```
+### Postprocess the `output` and print the result
```

Comment:
Maybe describe what is happening here in 1-2 sentences

Reply:
Or link back to the other tutorial

Reply:
Added!